### PR TITLE
Add zeropage inc instruction

### PR DESCRIPTION
**Pull request by robinlinden**

### CODE REVIEW

File: core/src/mos6502.cpp [hunk 1]
```diff
@@ -261,6 +261,23 @@ Pipeline Mos6502::parse_next_instruction() {
     case Instruction::NOP:
         result.push([]() { /* Do nothing. */ });
         break;
+    case Instruction::INC: {
+        if (opcode.address_mode == AddressMode::Zeropage) {
+            result.append(create_zeropage_addressing_steps());
+            result.push([=]() { tmp_ = mmu_->read_byte(effective_address_); });
+            result.push(
+                    [=]() { mmu_->write_byte(effective_address_, tmp_++); });
```
Comment (johnor) [inline core/src/mos6502.cpp]: I think some of the instructions we have implemented already skips similar extra writes. We should probably go through them again later and fix that :)

File: core/src/mos6502.cpp [hunk 2]
```diff
@@ -261,6 +261,23 @@ Pipeline Mos6502::parse_next_instruction() {
...
+            result.push([=]() {
+                set_zero(tmp_);
+                set_negative(tmp_);
+                mmu_->write_byte(effective_address_, tmp_);
+            });
+            break;
+        }
+        std::stringstream err;
```
Comment (johnor) [inline core/src/mos6502.cpp]: Should this go into a helper function? The same exception with the same error message is also thrown if the instruction is invalid, and should probably also be thrown in some other cases?

File: core/src/mos6502.cpp (hunk 1 above)
Comment (robinlinden) [reply core/src/mos6502.cpp]: You're right. :) Added an [issue](#107) for it.

File: core/src/mos6502.cpp (hunk 2 above)
Comment (robinlinden) [reply core/src/mos6502.cpp]: This isn't needed after merging master, so I deleted it. :D

**Pull request merged**